optuna/artifacts/_download.py

```diff
@@ -0,0 +1,31 @@
+from __future__ import annotations
+
+from optuna._experimental import experimental_func
+from optuna.artifacts._protocol import ArtifactStore
+
+
+BUFFER_SIZE = 1024 * 1024  # 1MB
+
+
+@experimental_func("4.0.0")
+def download_artifact(
+    artifact_store: ArtifactStore,
+    artifact_id: str,
+    file_path: str,
+) -> None:
+    """Download an artifact from the artifact store.
+
+    Args:
+        artifact_store:
+            An artifact store.
+        artifact_id:
+            The identifier of the artifact to download.
+        file_path:
+            A path to download the artifact to.
+    """
+    with artifact_store.open_reader(artifact_id) as reader, open(file_path, "wb") as writer:
+        while True:
+            chunk = reader.read(BUFFER_SIZE)
+            if not chunk:
```

Comment:
As `chunk` becomes `None` at end of line, it is better to use `chunk is None` because `chunk is None` is a stricter condition.
https://docs.python.org/ja/3/library/io.html#io.RawIOBase.read
```suggestion
            if chunk is None:  # `chunk` becomes `None` at EOL.
```

Reply:
@nabenabe0928 Thank you for your careful review.

Let me clarify my understanding of the following sentence.
Does this change always read the file with non-blocking mode?

> If 0 bytes are returned, and size was not 0, this indicates end of file. If the object is in non-blocking mode and no bytes are available, None is returned.

Reply:
@toshihikoyanase 
You are right, and sorry for the confusion.
I modified the suggestion as follows:

```suggestion
            if not chunk:  # `chunk` becomes an empty string at EOL.
```